Qfront/src/redux/departments/Departments.js

```diff
@@ -0,0 +1,33 @@
+import {getAllDepartments} from './actions';
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+    fetchedDepartments: [],
+    departmentsApiError: null,
+    loading: false,
+    success: null,
+};
+
+// Create departments reducer
+const departmentsSlice = createSlice({
+    name: "departments",
+    initialState,
+    reducers: {},
+    extraReducers: (builder) => {
+        builder.addCase(getAllDepartments.fulfilled, (state, action) => {
+            state.success = true;
+            state.fetchedDepartments = action.payload;
+            state.loading = false;
+        });
+        builder.addCase(getAllDepartments.pending, (state) => {
+            state.loading = true;
+        });
+        builder.addCase(getAllDepartments.rejected, (state, action) => {
+            state.success = false;
+            state.departmentsApiError = action.payload;
```

Comment:
What is the structure of the error? We only need the error message, and not the entire error object